internal/logger/logger_test.go

```diff
@@ -39,6 +45,114 @@ func BenchmarkLogger(b *testing.B) {
 	})
 }
 
+// mockKeyValues will return a slice of alternating keys and values of the
+// given length and the resulting JSON string.
+func mockKeyValuesB(b *testing.B, length int) KeyValues {
+	b.Helper()
+
+	keysAndValues := KeyValues{}
+	for i := 0; i < length; i++ {
+		keyName := fmt.Sprintf("key%d", i)
+		valueName := fmt.Sprintf("value%d", i)
+
+		keysAndValues.Add(keyName, valueName)
+	}
+
+	return keysAndValues
+}
+
+func BenchmarkIOSinkInfo(b *testing.B) {
+	keysAndValues := mockKeyValuesB(b, 100)
```

Comment:
Optional: No actual log messages emitted by the Go driver have 50 key/value pairs. Consider changing that value to 10 or 20 to more closely reproduce a real-world log message.